### PR TITLE
Print to console by default when not run with -daemon

### PR DESCRIPTION
**Pull request by practicalswift**
Cherry-picked ef6fa1c38e1bd115d1cce155907023d79da379d8 from the "up for grabs" PR: "Smarter default behavior for -printtoconsole" (#12689).

See previous review in #12689.

### CODE REVIEW

**Comment by fanquake** [comment]
@ryanofsky Could you re-review here.

**Comment by practicalswift** [comment]
Previous utACK:ers @MeshCollider and @jnewbery might want to take a look at this one too? :-)

**Comment by instagibbs** [comment]
concept ACK

File: src/init.cpp [hunk 1]
```diff
@@ -1216,13 +1227,12 @@ bool AppInitMain()
...
     CreatePidFile(GetPidFile(), getpid());
 #endif
-    if (gArgs.GetBoolArg("-shrinkdebugfile", logCategories == BCLog::NONE)) {
-        // Do this first since it both loads a bunch of debug.log into memory,
-        // and because this needs to happen before any other debug.log printing
-        ShrinkDebugFile();
-    }
-
     if (fPrintToDebugLog) {
+        if (gArgs.GetBoolArg("-shrinkdebugfile", logCategories == BCLog::NONE)) {
```
Comment (laanwj) [inline src/init.cpp]: Whoa - good catch to move this inside the `if (fPrintToDebugLog)`

**Comment by laanwj** [comment]
Thanks for reviving this, tested ACK 6a3b0d3d1aee1ab924f30b9910bc517c764917cd:
- `src/bitcoind -regtest` works as expected, now prints logging output
- `src/bitcoind -regtest -noprinttoconsole` successfully silences logging output
- `src/bitcoind -regtest -daemon` successfully silences logging output

File: test/functional/feature_logging.py [hunk 2]
```diff
@@ -15,21 +15,25 @@ def set_test_params(self):
         self.num_nodes = 1
         self.setup_clean_chain = True
 
+    def relative_log_path(self, name):
+        return os.path.join(self.nodes[0].datadir, "regtest", name)
```
Comment (MarcoFalke) [inline test/functional/feature_logging.py]: nit: IIRC `datadir` is absolute, so the `relative` in the method name is confusing.

File: src/init.cpp [hunk 3]
```diff
@@ -815,14 +815,25 @@ static std::string ResolveErrMsg(const char * const optname, const std::string&
...
+ * Note that this is called very early in the process lifetime, so you should be
+ * careful about what global state you rely on here.
+ */
 void InitLogging()
 {
-    fPrintToConsole = gArgs.GetBoolArg("-printtoconsole", false);
+    // Add newlines to the logfile to distinguish this execution from the last
+    // one; called before console logging is set up, so this is only sent to
+    // debug.log.
+    LogPrintf("\n\n\n\n\n");
```
Comment (jimpo) [inline src/init.cpp]: Hmm, won't it print these newlines even if `fPrintToDebugLog` ends up being false?

**Pull request merged**